src/Result.php

```diff
@@ -21,21 +21,35 @@ class Result {
      */
     public $number_of_matches_total = 0;
     /**
-     * @var array all posts that matched the query
+     * @var Post[] all posts that matched the query
      */
     public $posts = [];
 
     /**
+     * true if if one or multiple servers were too slow to respond within the maximum allowed query time
+     * false if all servers responded within the maximum allowed query time
+     * @var bool
+     */
+    public $incomplete_result;
+
+    /**
      * @return bool returns TRUE if this result includes all Posts that matched the query
      */
     public function all_results_returned() {
         return $this->number_of_matches_returned == $this->number_of_matches_total;
     }
 
+    /**
+     * @return bool returns whether the result is incomplete or not
+     */
+    public function incomplete() {
```

Comment:
I don't think there is any reason for this method since there already is a publicly accessible `$incomplete_result` variable.